airbyte-commons/src/main/java/io/airbyte/commons/json/JsonPaths.java

```diff
@@ -0,0 +1,278 @@
+
+      @Override
+      public JsonProvider jsonProvider() {
+        return jsonProvider;
+      }
+
+      @Override
+      public MappingProvider mappingProvider() {
+        return mappingProvider;
+      }
+
+      @Override
+      public Set<Option> options() {
+        /*
+         * All JsonPath queries will return a list of values. This makes parsing the outputs much easier. In
+         * cases where it is not a list, helpers in this class can assert that. See
+         * https://github.com/json-path/JsonPath in the JsonPath documentation.
+         */
+        return EnumSet.of(Option.ALWAYS_RETURN_LIST);
+      }
+
+    });
+  }
+
+  /*
+   * This version of the JsonPath Configuration object allows queries to return to the path of values
+   * instead of the values that were found.
+   */
+  private static final Configuration GET_PATHS_CONFIGURATION = Configuration.defaultConfiguration().addOptions(Option.AS_PATH_LIST);
+
+  /**
+   * Attempt to validate if a string is a valid JSONPath string. This assertion does NOT handle all
+   * cases, but at least a common on. We can add to it as we detect others.
+   *
+   * @param jsonPath - path to validate
+   */
+  public static void assertIsJsonPath(final String jsonPath) {
+    Preconditions.checkArgument(jsonPath.startsWith("$"));
+  }
+
+  /**
+   * Attempt to detect if a JSONPath query could return more than 1 value. This assertion does NOT
+   * handle all cases, but at least a common on. We can add to it as we detect others.
+   *
+   * @param jsonPath - path to validate
+   */
+  public static void assertIsSingleReturnQuery(final String jsonPath) {
+    Preconditions.checkArgument(!jsonPath.contains("*"), "Cannot accept paths with wildcards because they may return more than one item.");
+  }
+
+  /**
+   * Given a JSONPath, returns all the values that match that path.
+   *
+   * e.g. for this object: { "alpha": [1, 2, 3] }, if the input JSONPath were "$.alpha[*]", this
+   * function would return: [1, 2, 3].
+   *
+   * @param json - json object
+   * @param jsonPath - path into the json object. must be in the format of JSONPath.
+   * @return all values that match the input query
+   */
+  public static List<JsonNode> getValues(final JsonNode json, final String jsonPath) {
+    return getInternal(Configuration.defaultConfiguration(), json, jsonPath);
+  }
+
+  /**
+   * Given a JSONPath, returns all the path of all values that match that path.
+   *
+   * e.g. for this object: { "alpha": [1, 2, 3] }, if the input JSONPath were "$.alpha[*]", this
+   * function would return: ["$.alpha[0]", "$.alpha[1]", "$.alpha[2]"].
+   *
+   * @param json - json object
+   * @param jsonPath - path into the json object. must be in the format of JSONPath.
+   * @return all paths that are present that match the input query
+   */
+  public static List<String> getPaths(final JsonNode json, final String jsonPath) {
```

Comment:
Nit: Should we enforce it being a` Set`? The path should be uniq

Reply:
yeah. good point.

Reply:
actually upon further thought i think the list helps avoid bugs due to not determinitic ordering behavior. added to the java docs to explain it.